### PR TITLE
Password iteration

### LINKED ISSUE #386: Password Page: UI Iteration
(Placeholder) 

Exploration to elevate the look and feel of the password page with the following objectives: 

1. When not customized, password page can be a great brand touchpoint for Shopify ( with messaging like "A new merchant is about to fulfill their dreams with Shopify") - would partner with Marketing / Comms to align on placeholder copy / assets

2. When customized, this page should feel exiting, like a [pre-launch page](https://startupfashion.com/5-great-pre-launch-landing-pages-to-inspire-you/) and do a good job at teasing a merchant's brand. 

Note: Would still be important for this page to not require image assets, as not all merchants have access to quality ones from the start. 

https://videobin.shopify.io/v/26mGE

### PR DESCRIPTION
**Pull request by sofiamatulis**
**Why are these changes introduced?**

Fixes #386 .

https://www.figma.com/file/JcYdswrk2JhqBTGBVWPM7J/Dawn-2.0?node-id=1090%3A19973

**What approach did you take?**

We are adding a new section (name TBD) to the password template. It is only available in the password template. It will be there as a smart default for a new merchant signing up and will be available as a new section for existing merchants (because of that, we are allowing for the section to be added multiple times which is what we already have for the other email sign up and other sections as well)

This section seems to be a combination of Image Banner and Email Sign up so I tried to reuse some of its CSS to avoid duplicating it. 

**Other considerations**

The original idea was to update the existing email signup section to behave differently on the Password template. That would mean it's a breaking change, since merchants may have that section already on their password template (and it's a default as well). We didnt want to change the current section too because it has the full width setting, which is something we didnt want on the new email sign up section, but we couldnt remove it without it being a breaking change.

We are able to ensure that this **new** section is on the password template only, but we wanted to ensure the existing email sign up section is also available in all of the templates (including password), since we don't want to list it to be everywhere **but** on the password page (and removing this section would also be a breaking change)

Because we want the default image to be showing, we added a checkbox "Show image" to ensure that the merchant can still turn that off if desired.

Please note that there are PRs that update the banner width and height: https://github.com/Shopify/dawn/pull/663 https://github.com/Shopify/dawn/pull/695

**Expectations**

- Should be on the password template ONLY
- Existing merchants should be able to add a section
- New merchants should see it as the default
- Should be able to add multiple email sign up sections
- Background should be turned on by default
- Previous email sign up should be removed as a default for new merchants

**Demo links**

- [Editor](https://os2-demo.myshopify.com/admin/themes/126449909782/editor)

**Checklist**
- [ ] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [ ] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [ ] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [ ] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [ ] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

### CODE REVIEW

**Comment by Oliviammarcello** [comment]
Looking great so far!

Since the SVG will be updated my comments for that will come at a later date. 
- [ ] Can we set a minimum height on the images on mobile to avoid this?[Screenshot](https://screenshot.click/21-12-b08er-mfcsq.png)
- [ ] When the content doesn't fill the screen, can it align to the center? [Screenshot](https://screenshot.click/21-22-a5lym-87x2j.png)
- [ ] The paragraph text should move up closer to the title. The spacing that is between the paragraph and input field should switch with the spacing between the title and paragraph on both desktop and mobile. [Screenshot. ](https://screenshot.click/21-20-zmptk-thalp.png)
- [ ] On mobile, when the adapt to image banner height setting is not enabled, can the image width fill the available space? [Screenshot. ](https://screenshot.click/21-21-ddax9-6ovoy.png)
- [ ] - [ ] The adapt to image height checkbox does not seem to change the height on desktop

Let me know if you would like to pair on anything, thanks!

**Comment by sofiamatulis** [comment]
> Looking great so far!
> 
> Since the SVG will be updated my comments for that will come at a later date.
> 
> * [ ]  Can we set a minimum height on the images on mobile to avoid this?[Screenshot](https://screenshot.click/21-12-b08er-mfcsq.png)
> * [ ]  When the content doesn't fill the screen, can it align to the center? [Screenshot](https://screenshot.click/21-22-a5lym-87x2j.png)
> * [x]  The paragraph text should move up closer to the title. The spacing that is between the paragraph and input field should switch with the spacing between the title and paragraph on both desktop and mobile. [Screenshot. ](https://screenshot.click/21-20-zmptk-thalp.png)
> * [ ]  On mobile, when the adapt to image banner height setting is not enabled, can the image width fill the available space? [Screenshot. ](https://screenshot.click/21-21-ddax9-6ovoy.png)
> * [ ]  The adapt to image height checkbox does not seem to change the height on desktop
> 
> Let me know if you would like to pair on anything, thanks!

 @Oliviammarcello I will wait for the banner height PR to be merged to ensure I have the most up to date styles and then address the other feedback :)

https://github.com/Shopify/dawn/pull/695

For the elements in the banner box, since they are blocks and re-ordeable I have followed what we do for banner: https://screenshot.click/04-48-ad26v-fzlro.mp4

**Comment by gretchen-willenberg** [comment]
> * When the content doesn't fill the screen, can it align to the center? [Screenshot](https://screenshot.click/21-22-a5lym-87x2j.png)

Agree with this when copy is under 3 lines. If merchants have the ability to enter a paragraph, we should consider forcing left (or right align depending on language) for readability best practices. 
[Why you should never center align paragraph text](https://uxmovement.com/content/why-you-should-never-center-align-paragraph-text/#:~:text=Left%20aligned%20text%20is%20easier,line%20begins%20to%20continue%20reading.&text=Centered%20text%20is%20best%20used%20for%20headlines%20and%20short%20lines%20of%20text)
[Text alignment best practices: make your UI more effective](https://blog.prototypr.io/text-alignment-best-practises-c4114daf1a9b)


Other copy, like `Enter using password` is carryover from Debut; don't have data or other reason to revise now.

**Comment by sofiamatulis** [comment]
> Other copy, like `Enter using password` is carryover from Debut; don't have data or other reason to revise now.

@gretchen-willenberg In terms of copy, I meant the new settings copy for this new section: https://github.com/Shopify/dawn/pull/698/files#diff-8634bdb8283e62e6df834e80736461b1c05b2d6d669fde29e2fe1b9b7e86ed87R1712-R1792

If that looks good, I can request translations 🙏

File: sections/email-signup-banner.liquid
Comment (ludoboludo) [inline sections/email-signup-banner.liquid:25]: I think here we can cut out some repetitions: 

```suggestion
  @media screen and (max-width: 749px) {
    #Banner-{{ section.id }}:not(.banner--mobile-bottom) .banner__content::before {
      padding-bottom: {{ 1 | divided_by: section.settings.image.aspect_ratio | times: 100 }}%;
      content: '';
      display: block;
    }
  }

  #Banner-{{ section.id }}::before,
  #Banner-{{ section.id }} .banner__media::before {
    padding-bottom: {{ 1 | divided_by: section.settings.image.aspect_ratio | times: 100 }}%;
    content: '';
    display: block;
  }
```

File: sections/email-signup-banner.liquid
Comment (ludoboludo) [inline sections/email-signup-banner.liquid:33]: I wonder, can you have an if statement within the `style` tags ? Cause maybe you could have only one. Something like this: 
```liquid
{%- style -%}
  #Banner-{{ section.id }}::after {
    opacity: {{ section.settings.image_overlay_opacity | divided_by: 100.0 }};
  }

  {%- if section.settings.adapt_height_image and section.settings.image != blank -%}
    @media screen and (max-width: 749px) {
      #Banner-{{ section.id }}:not(.banner--mobile-bottom) .banner__content::before {
        padding-bottom: {{ 1 | divided_by: section.settings.image.aspect_ratio | times: 100 }}%;
        content: '';
        display: block;
      }
    }

    #Banner-{{ section.id }}::before,
    #Banner-{{ section.id }} .banner__media::before {
      padding-bottom: {{ 1 | divided_by: section.settings.image.aspect_ratio | times: 100 }}%;
      content: '';
      display: block;
    }
  {%- endif -%}
{%- endstyle -%}
```

It seemed to work when tested locally.

File: sections/email-signup-banner.liquid
Comment (ludoboludo) [inline sections/email-signup-banner.liquid:47]: There was a recent update on images srcset and they now all include another line at the end: `{{ section.settings.image | img_url: 'master' }} {{ section.settings.image.width }}w`.

[PR for reference](https://github.com/Shopify/dawn/pull/668)

File: sections/email-signup-banner.liquid
Comment (ludoboludo) [inline sections/email-signup-banner.liquid:57]: Since those two SVG files are only used here, I think we could have the code right here rather than in a snippet.

File: locales/en.default.schema.json
Comment (ludoboludo) [inline locales/en.default.schema.json:1742]: I would expect the color scheme to still influence the font color even when the text box background is removed.

Here for example it should be black:

File: assets/section-password.css
Comment (ludoboludo) [inline assets/section-password.css:332]: Id maybe go with less generic class name 🤔 At least one that reflects that it's on the password page.

File: assets/section-password.css
Comment (ludoboludo) [inline assets/section-password.css:352]: There seem to be an issue with links from the text rich text blocks. They're not using the right colors: 


It's not only for this sections, the original email signup one has it too. 

I think calling the `component-rte.css` should do the trick.

File: sections/email-signup-banner.liquid
Comment (ludoboludo) [inline sections/email-signup-banner.liquid:71]: I think here we're supposed to keep the background color on the input. Though should that color be dictated by the color scheme chosen ?

File: assets/section-password.css
Comment (ludoboludo) [inline assets/section-password.css:327]: When some images are not super big and with the text box over them, it can leave some white gap between the section and footer. 

Should we keep `flex-grow` on the `main` element ? 🤔

File: locales/en.default.schema.json
Comment (Oliviammarcello) [reply locales/en.default.schema.json:1742]: Because of the opacity overlay, the text has to remain white.

File: locales/en.default.schema.json
Comment (sofiamatulis) [reply locales/en.default.schema.json:1737]: @ludoboludo This is behaving like the image banner: when I toggle the `show text box on desktop` I dont see any changes on mobile 🤔 . Do you have a video of what you mean exactly?

Video of current experience not affecting mobile: https://screenshot.click/05-31-yiz3l-1acmv.mp4

File: sections/email-signup-banner.liquid
Comment (sofiamatulis) [reply sections/email-signup-banner.liquid:57]: @ludoboludo If you open `password-background` because it's a detailed SVG youll see the file is quite large so I dont think itll be very readable

File: assets/section-password.css
Comment (sofiamatulis) [reply assets/section-password.css:332]: Do you have any suggestions? 😅 What about `password-banner-emailsignup-background`. I feel like it's a bit too long

File: assets/section-password.css
Comment (sofiamatulis) [reply assets/section-password.css:352]: Good catch! Fixed

File: assets/section-password.css
Comment (sofiamatulis) [reply assets/section-password.css:327]: @Oliviammarcello @ludoboludo The reason this was added was to ensure the footer always stays at the bottom (I will try to find the PR and the discussion/reference). Otherwise this would happen: 

This is just this one scenario of a shorter image and only one section.

Video: https://screenshot.click/05-42-3bglx-zh3lg.mp4

Options:

1) remove and have the footer go up (it may look odd because the page on mobile looks short)
2) keep it and ensure that the footer stays at the bottom

File: sections/email-signup-banner.liquid
Comment (sofiamatulis) [reply sections/email-signup-banner.liquid:72]: I think it makes sense to keep it consistent with the title/paragraph in case the merchant adds a darker image or the overlay. Olivia's comment for reference too: https://github.com/Shopify/dawn/pull/698#discussion_r722222247

File: locales/en.default.schema.json
Comment (sofiamatulis) [reply locales/en.default.schema.json:1719]: @ludoboludo In terms of content, I will tag @Oliviammarcello @gretchen-willenberg - not sure if we should do that since we dont for image with text and image banner and I think it makes sense to keep it consistent

In terms of the heights, I will rebase this off of Ken's banner height PR updates so we can have the most up to date code.

File: locales/en.default.schema.json
Comment (ludoboludo) [reply locales/en.default.schema.json:1742]: Ah i see. And on the other hand, if you are using a light coloured image you will have to use the overlay opacity or text box background in order to see the text properly. Got it 👍

File: sections/email-signup-banner.liquid
Comment (ludoboludo) [reply sections/email-signup-banner.liquid:52]: ah! good to know 👍

**Comment by gretchen-willenberg** [comment]
> > Other copy, like `Enter using password` is carryover from Debut; don't have data or other reason to revise now.
> 
> @gretchen-willenberg In terms of copy, I meant the new settings copy for this new section: https://github.com/Shopify/dawn/pull/698/files#diff-8634bdb8283e62e6df834e80736461b1c05b2d6d669fde29e2fe1b9b7e86ed87R1712-R1792
> 
> If that looks good, I can request translations 🙏

Thanks for clarifying--since the initial launch on Aug 31, I haven't been involved with Dawn settings content so I lack context on revisions or further development. If this content is from Olivia and Melissa, I defer to their work, so it can be translated.

**Comment by sofiamatulis** [comment]
> > > Other copy, like `Enter using password` is carryover from Debut; don't have data or other reason to revise now.
> > 
> > 
> > @gretchen-willenberg In terms of copy, I meant the new settings copy for this new section: https://github.com/Shopify/dawn/pull/698/files#diff-8634bdb8283e62e6df834e80736461b1c05b2d6d669fde29e2fe1b9b7e86ed87R1712-R1792
> > If that looks good, I can request translations 🙏
> 
> Thanks for clarifying--since the initial launch on Aug 31, I haven't been involved with Dawn settings content so I lack context on revisions or further development. If this content is from Olivia and Melissa, I defer to their work, so it can be translated.

@melissaperreault @Oliviammarcello When you have a chance, do you mind taking a look at the content to see if it looks good so I can request translations :)

File: assets/section-password.css
Comment (sofiamatulis) [reply assets/section-password.css:327]: I think it actually makes sense to have it below because youd want the footer not to be floating and have a lot of white space below all the content 🤔 it makes the page feel short I think which is a weird experience on mobile (also this is an edge case for a short image). Once the heights get adjusted too (Ken's PR), well have a different experience so itll be good to have those changes before

https://screenshot.click/05-01-u33m1-5mu8l.png

**Review by ludoboludo — COMMENTED** [review]
Minor comments, im gonna wait until we hear back from designers and get that PR from Ken merged to re review.

File: assets/section-password.css
Comment (ludoboludo) [reply assets/section-password.css]: See on wider screens even images that are fairly big might end up showing something like this which is kinda "meh" I find: [screenshot](https://screenshot.click/21-10-67s2u-drqbb.png)

Comment (sofiamatulis) [reply assets/section-password.css]: Yea, this definitely looks better if it's a shorter image ... 🤔 Once I rebase we can retest these scenarios

File: config/settings_data.json
Comment (sofiamatulis) [reply config/settings_data.json:118]: Yes! I will revert before the merge

File: sections/email-signup-banner.liquid
Comment (wiktoriaswiecicka) [reply sections/email-signup-banner.liquid:66]: I would recommend to keep the input transparent to match styling on other sections and let merchant adjust the background.

File: sections/email-signup-banner.liquid
Comment (KaichenWang) [reply sections/email-signup-banner.liquid:244]: Ah yes i see it! I was looking in the wrong place 😅 sorry!

File: sections/email-signup-banner.liquid
Comment (sofiamatulis) [reply sections/email-signup-banner.liquid:135]: Just a quick update: banner has styles to have a min-height of 72rem on desktop even with the `adapt height to image` checked off. I think this is to avoid a very short image since it's still a banner. But once I have the updated styles, we can retest. I compared and it's behaving exactly like the banner

File: sections/email-signup-banner.liquid
Comment (melissaperreault) [reply sections/email-signup-banner.liquid:29]: Have we considered offering this type of flexibility to manage the height?

cc. @wiktoriaswiecicka @nicklepine

Comment (sofiamatulis) [reply sections/email-signup-banner.liquid:29]: @melissaperreault @wiktoriaswiecicka @nicklepine I am not sure if that is the reason, but thinking about it I think this section is already a combination of banner + email sign up so it already has more settings. Screenshot: https://screenshot.click/13-23-hl7ry-5d3c2.png

So I think that adding more settings, may become a bit much? 🤔 but not sure if that was the reason behind it.

We also have the checkbox `adapt to image height` so the merchant could choose between the two

**Comment by melissaperreault** [comment]
Sharing some observations:

- [ ] There is [some empty space to the right](https://screenshot.click/21-10-08epx-933oi.png) on small devices
- [ ] Is it expected to [have the visual not full bleed](https://screenshot.click/21-10-kb5c3-uk0ut.png) at all time?

File: sections/email-signup-banner.liquid
Comment (sofiamatulis) [reply sections/email-signup-banner.liquid:29]: @melissaperreault To play devil's advocate, you're right 🤔. we dont know what kind of image they are uploading so more control does sound like it would be beneficial. I wonder if we have any data on how they are currently using the banner section? 

I am curious if it was a consideration as well

Comment (melissaperreault) [reply sections/email-signup-banner.liquid:29]: This is the main reason why we introduced Image height setting to have more flexibility to accommodate various needs. https://github.com/Shopify/dawn/issues/552 solved by https://github.com/Shopify/dawn/pull/695, we can talk about the risk of providing or not for this particular template. Perhaps I am missing some expectations from that template.

Comment (sofiamatulis) [reply sections/email-signup-banner.liquid:29]: Yea 🤔 I can see it going both ways and it can be done as a fast follow as well.

Something to keep in mind is that this section is only available on the password page and it's the combination with the email sign up, so maybe the expectations are a bit different? We can discuss what makes the most sense

File: locales/en.default.schema.json
Comment (sofiamatulis) [reply locales/en.default.schema.json:1739]: I added the same aspect ratio's content that we had on banner as per our meeting today to copy over the settings for banner

**Comment by sofiamatulis** [comment]
@wiktoriaswiecicka @melissaperreault @nicklepine 

I updated 
- SVG stretches to full width on desktop and mobile to avoid empty space on the sides
- Footer stays always at the bottom
- New height settings were added

If the new settings content looks good, let me know and I will request translations :)

File: assets/section-email-signup-banner.css
Comment (KaichenWang) [inline assets/section-email-signup-banner.css:22]: Are these styles needed? `email-signup-banner-background` already includes the same CSS properties.

File: locales/en.default.schema.json
Comment (melissaperreault) [inline locales/en.default.schema.json:1799]:
```suggestion
        }
```
Is there missing a coma here?

File: locales/en.default.schema.json
Comment (KaichenWang) [reply locales/en.default.schema.json:1762]: Out of scope for this PR, but I agree with Ludo here. We should not hardcode color values here. I think the color system could work perfectly fine with an overlay opacity slider as well.

File: locales/en.default.schema.json
Comment (melissaperreault) [inline locales/en.default.schema.json:1786]: Let's move this setting below the `Banner height` setting to improve the order and align [with Image banner section.](https://screenshot.click/21-10-g2o6i-5qz78.png)

File: locales/en.default.schema.json
Comment (melissaperreault) [inline locales/en.default.schema.json:1788]:
```suggestion
          "label": "Banner height",
```
Based on Gretchen's recommendation, I'll do a final pas on content to align image banner shortly too!

**Review by KaichenWang — DISMISSED** [review]
Looks great!

File: locales/en.default.schema.json
Comment (sofiamatulis) [reply locales/en.default.schema.json:1799]: That is the last item so no coma needed :)

File: locales/en.default.schema.json
Comment (melissaperreault) [reply locales/en.default.schema.json:1786]: Oups, let's forget about this and instead include the option inside the select list!

File: assets/section-email-signup-banner.css
Comment (sofiamatulis) [reply assets/section-email-signup-banner.css:22]: Good catch!

**Review by melissaperreault — DISMISSED** [review]
Thank you Sofia for the updates! Added a few notes but works perfectly! Looking really good, once the translations are back in I'll approve again!

File: locales/en.default.schema.json
Comment (melissaperreault) [inline locales/en.default.schema.json:1787]:
```suggestion
            "label": "Adapt to image"
```
I believe this is the right language used elsewhere in the theme!

File: locales/en.default.schema.json
Comment (melissaperreault) [inline locales/en.default.schema.json:873]:
```suggestion
          "info": "Used for text box."
```
Just missing the period for info text.

File: sections/email-signup-banner.liquid
Comment (ludoboludo) [inline sections/email-signup-banner.liquid:11]: This isn't working as it's not checking the right setting I think:
```suggestion
  {%- if section.settings.image_height ==  `adapt-image` and section.settings.image != blank -%}
```

File: assets/section-email-signup-banner.css
Comment (ludoboludo) [inline assets/section-email-signup-banner.css:35]: A couple things I'm wondering about the styling of the section: 

- When the section is set to large and is the only section, should we add a `height: 100%` to prevent [this white space](https://screenshot.click/21-10-3f45t-3xt7x.png) from showing? 
- When the section isn't using an image and is the only section, should we center it vertically by making the parent height `100%` ? [screenshot](https://screenshot.click/21-10-t7qf7-gg6t8.png)

File: sections/email-signup-banner.liquid
Comment (sofiamatulis) [reply sections/email-signup-banner.liquid]: Good catch , Ludo!

**Comment by sofiamatulis** [comment]
> this video

@ludoboludo This is a really good catch. Not sure if its something we should change on this PR, because I just noticed the same happens for the banner so wed have to update both. But also not sure if adding more settings would be also the best solution. Not sure if itd be too many settings? 🤔

**Review by ludoboludo — APPROVED** [review]

**Comment by KaichenWang** [comment]
> @KaichenWang This is actually expected and happens for the banner section too 🤔 I believe it's because if you dont have "enough height" it wouldnt like it was aligned to the bottom or top

Thanks for the context!

**Pull request merged**
